pkg/docker/docker.go

```diff
@@ -222,3 +229,31 @@ func ExtractResult(reader io.Reader, command []string) *Result {
 	}
 	return &result
 }
+
+func checkImageExist(ctx context.Context, cli *client.Client, image string, notag bool) bool {
```

Comment:
There are many conditionals in this and unit tests will help understand better.

Reply:
@ayushjn20 Did you add unit tests?

Reply:
Done